src/lib/description-helpers/endpoint-parsers.ts

```diff
@@ -0,0 +1,303 @@
+import type { Schema } from "genson-js";
+import { Endpoint } from "../../utils/types";
+import tokenizer from "gpt-tokenizer";
+import { ChatMessage } from "../../ui/helpers/count-tokens"; 
+
+export type Result = { [key: string]: any };
+
+// Helper function to convert method details into a string
+export function methodDetailsToString(method: any): string {
+
+  const queryParametersString = method.queryParameters
+    ? Object.entries(method.queryParameters.params?.properties || {})
+        .map(([paramName, paramSchema]) => {
+          const example = method.queryParameters.mostRecent?.[paramName];
+          const exampleString = example !== undefined ? JSON.stringify(example) : 'No recent example';
+          const paramType = (paramSchema as { type?: string }).type || 'unknown';
+          return `${paramName}: ${paramType}. ${exampleString}`;
+        })
+        .join('\n')
+    : 'No query parameters';
+
+
+  const requestString = method.request
+    ? Object.entries(method.request as { [mediaType: string]: any }).map(([mediaType, req]) => {
+        const parsedRequest = req.mostRecent
+          ? JSON.stringify(findExamplesFromJSON(req.mostRecent))
+          : 'No recent example';
+        return `${mediaType}: Summarised Example Request: ${parsedRequest}`;
+      }).join('\n')
+    : 'No request info';
+
+  const responseString = method.response
+    ? Object.entries(method.response as { [statusCode: string]: { [mediaType: string]: any } }).map(([statusCode, responses]) => {
+        return `${statusCode}: ` + Object.entries(responses).map(([mediaType, res]) => {
+          const mostRecent = res.mostRecent
+            ? JSON.stringify(findExamplesFromJSON(res.mostRecent))
+            : 'No recent sample';
+          return `${mediaType}: Summarised Example Response: ${mostRecent}`;
+        }).join('\n');
+      }).join('\n')
+    : 'No response info';
+
+  return `Query Parameters: \n${queryParametersString}\nRequest:\n${requestString}\nResponse:\n${responseString}`;
+}
+
+
+export function endpointToString(endpoint: Endpoint): string {
+  const partsString = endpoint.parts.map(part => `${part.part} (type: ${part.type})`).join(', ');
```

Comment:
Why include both `parts` and `pathname`? Isn't one a subset of the other?

Reply:
the parts string also includes the type